Assets/Scripts/Construction/PlayerConstructionHandler.cs

```diff
@@ -22,9 +22,15 @@ public class PlayerConstructionHandler : MonoBehaviour, DefaultInputActions.IBui
 
         void LateUpdate()
         {
-            ProcessPlacement();
+            if (currentBuildingInfo != null)
+            {
+                InitiatePlacement(currentBuildingInfo);
+                ProcessPlacement();
+            }
         }
 
+        public void SetBuilding(BuildingInfo buildingInfo) { currentBuildingInfo = buildingInfo; }
```

Comment:
Can we just use a setter? And even if we don't want to make it a setter we want to have this func like all others example in suggestion.
```suggestion
        public void SetBuilding(BuildingInfo buildingInfo)
        {
                currentBuildingInfo = buildingInfo;
        }
```